### PR TITLE
Browse exiting temp

### PR DESCRIPTION
**Pull request by adithyadinesh0412**

### CODE REVIEW

File: src/services/resource.js [hunk 1]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+				data: [],
+				count: 0,
+			}
+			const type = query[common.TYPE] ? query[common.TYPE] : ''
+			const search = searchText != '' ? searchText : ''
+			let resources = {}
+			// consumption side if set to self , only resources published with in SCP will be showed
+			// If it has any value other than self , the result will be combination of resources from the coupled service and from SCP.
+			if (process.env.CONSUMPTION_SERVICE != common.SELF) {
+				resources = await interfaceRequests.browseExistingList(type, organization_id, token, search)
```
Comment (priyanka-TL) [inline src/services/resource.js]: @adithyadinesh0412 can we change this to externalResources

File: src/services/resource.js [hunk 2]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+	 * @param {Integer} pageSize -  Used to limit the data. Used for pagination . If value is not passed, by default it will be 100
+	 * @returns {Object} - Response contain object of user details
+	 */
+	static async browseExistingList(organization_id, token, query, searchText = '', pageNo, pageSize) {
+		try {
+			let result = {
+				data: [],
+				count: 0,
+			}
+			const type = query[common.TYPE] ? query[common.TYPE] : ''
```
Comment (priyanka-TL) [inline src/services/resource.js]: @adithyadinesh0412 rename type to resourceType

File: src/services/resource.js [hunk 3]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+				organization_id,
+				status: common.RESOURCE_STATUS_PUBLISHED,
+				published_id: null,
+			}
+			if (type) filterQiuery.type = type
+			if (search)
+				filterQiuery.title = {
+					[Op.iLike]: `%${search}%`,
+				}
+			const selfResources = await resourceQueries.findAll(filterQiuery, [
```
Comment (priyanka-TL) [inline src/services/resource.js]: @adithyadinesh0412 change to internalResources

File: src/services/resource.js [hunk 4]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+				'id',
+				'title',
+				'type',
+				'created_by',
+				'created_at',
+			])
+
+			const combinedData = [
```
Comment (priyanka-TL) [inline src/services/resource.js]: @adithyadinesh0412 change combinedData to aggregatedResources

File: src/services/resource.js [hunk 5]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+				...(resources?.success && resources?.data?.result?.data?.length ? resources.data.result.data : []),
+				...(selfResources.length ? selfResources : []),
+			]
+
+			if (combinedData.length > 0) {
+				// construct sort object
+				const sort = await this.constructSortOptions(query.sort_by, query.sort_order)
```
Comment (priyanka-TL) [inline src/services/resource.js:1231]: @adithyadinesh0412 change sort to sortedResources

File: src/services/resource.js [hunk 6]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+				// sort the array
+				const sortedData = utils.sort(combinedData, sort)
```
Comment (priyanka-TL) [inline src/services/resource.js]: @adithyadinesh0412 change sortedData to sortedResources

File: src/services/resource.js [hunk 7]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+				// data after applying pagenation
+				const paginatedDate = utils.paginate(sortedData, pageNo, pageSize)
```
Comment (priyanka-TL) [inline src/services/resource.js]: @adithyadinesh0412 change paginatedDate to paginatedResources

File: src/services/resource.js [hunk 8]
```diff
@@ -1176,4 +1177,104 @@ module.exports = class resourceHelper {
...
+							const createdBy = resource.created_by
+							return !isNaN(createdBy) && !isNaN(parseFloat(createdBy)) ? +createdBy : createdBy
+						})
+					),
+					[null, undefined, '']
+				)
+				// fetch the user details from user service with creatorId
+				const userDetails = await this.fetchUserDetails(uniqueCreatorIds)
+
+				let finalResponse = []
```
Comment (priyanka-TL) [inline src/services/resource.js]: @adithyadinesh0412 change to finalResources

**Review by priyanka-TL — APPROVED** [review]

File: src/.env.sample [hunk 9]
```diff
@@ -166,4 +166,14 @@ RESOURCE_AUTO_SAVE_TIMER="30000"
...
+
+# Interface service host address 
+INTERFACE_SERVICE_HOST=http://localhost:3567
+
+# Interface service base URL 
+INTERFACE_BASE_URL=/interface/
+
+# set the end point of interface service to fetch the resources list from the consumption side
+# mandatory only if CONSUMPTION_SERVICE is not self
+BROWSE_EXISTING_END_POINT=v1/internal/fetchResources
```
Comment (aks30) [inline src/.env.sample]: @adithyadinesh0412 this need not be an env config

File: src/constants/endpoints.js [hunk 10]
```diff
@@ -10,4 +10,7 @@ module.exports = {
 		? process.env.USER_PROFILE_DETAILS_ENDPOINT
 		: 'v1/user/read',
 	VALIDATE_SESSIONS: 'v1/account/validateUserSession',
+	BROWSE_EXISTING_END_POINT: process.env.BROWSE_EXISTING_END_POINT
```
Comment (aks30) [inline src/constants/endpoints.js]: @adithyadinesh0412 this will not be via env

File: src/controllers/v1/resource.js [hunk 11]
```diff
@@ -37,6 +37,15 @@ module.exports = class Resource {
 					req.pageNo,
 					req.pageSize
 				)
+			} else if (req.query[common.LISTING] === common.PAGE_STATUS_BROWSE_EXISTING) {
```
Comment (aks30) [inline src/controllers/v1/resource.js]: @adithyadinesh0412 this has to be a new API

**Review by aks30 — CHANGES_REQUESTED** [review]
Reviewed on 30 Aug

File: src/.env.sample (hunk 9 above)
Comment (adithyadinesh0412) [reply src/.env.sample]: removed from env.

File: src/constants/endpoints.js (hunk 10 above)
Comment (adithyadinesh0412) [reply src/constants/endpoints.js]: removed from env

**Pull request merged**